python/lsst/validate/drp/checkAstrometryPhotometry.py

```diff
@@ -0,0 +1,514 @@
+        print("Median astrometric scatter %.1f %s is larger than reference : %.1f %s " %
+              (astromScatter, "mas", medianRef, "mas"))
+    if match < matchRef:
+        print("Number of matched sources %d is too small (shoud be > %d)" % (match, matchRef))
+
+    return astromScatter
+
+
+def checkPhotometry(mag, mmagrms, dist, match,
+                    good_mag_limit=19.5,
+                    medianRef=100, matchRef=500):
+    """Print out the astrometric scatter for all stars, and for good stars.
+
+    Inputs
+    ------
+    mag : list or numpy.array
+        Average magnitudes of each star
+    mmagrms ; list or numpy.array
+        Magnitude RMS of the multiple observation of each star.
+    dist : list or numpy.array
+        Distances between successive measurements of one star
+    match : int
+        Number of stars matched.
+
+    good_mag_limit : float, optional
+        Minimum average brightness (in magnitudes) for a star to be considered.
+    medianRef : float, optional
+        Median reference astrometric scatter in millimagnitudes
+    matchRef : int, optional
+        Should match at least matchRef stars.
+
+    Returns
+    -------
+    float
+        The photometry scatter (RMS, millimag) for all good stars.
+
+    Notes
+    -----
+    The scatter and match defaults are appropriate to SDSS are stored here.
+    For SDSS, stars with mag < 19.5 should be completely well measured.
+    This limit is a band-dependent statement most appropriate to r.
+    """
+
+    print("Median value of the photometric scatter - all magnitudes: %.3f %s" %
+          (np.median(mmagrms), "mmag"))
+
+    bright = np.where(np.asarray(mag) < good_mag_limit)
+    photoScatter = np.median(np.asarray(mmagrms)[bright])
+    print("Photometric scatter (median) - mag < %.1f : %.1f %s" %
+          (good_mag_limit, photoScatter, "mmag"))
+
+    if photoScatter > medianRef:
+        print("Median photometric scatter %.3f %s is larger than reference : %.3f %s " % (photoScatter, "mmag", medianRef, "mag"))
+    if match < matchRef:
+        print("Number of matched sources %d is too small (shoud be > %d)" % (match, matchRef))
+
+    return photoScatter
+
+
+def printPA2(gv, magKey):
+    """Calculate and print the calculated PA2 from the LSST SRD from a groupView."""
+
+    pa2 = calcPA2(gv, magKey)
+    print("minimum: PF1=%2d%% of diffs more than PA2 = %4.2f mmag (target is PA2 < 15 mmag)" % 
+          (pa2.PF1['minimum'], pa2.minimum))
+    print("design:  PF1=%2d%% of diffs more than PA2 = %4.2f mmag (target is PA2 < 15 mmag)" % 
+          (pa2.PF1['design'], pa2.design))
+    print("stretch: PF1=%2d%% of diffs more than PA2 = %4.2f mmag (target is PA2 < 10 mmag)" % 
+          (pa2.PF1['stretch'], pa2.stretch))
+
+def repoNameToPrefix(repo):
+    """Generate a base prefix for plots based on the repo name.
+
+    Examples
+    --------
+    >>> repoNameToPrefix("a/b/c")
+    "a_b_c"
+    >>> repoNameToPrefix("/bar/foo/")
+    "bar_foo"
```

Comment:
I don't think this works as advertised. I get:
```python
>>> repoNameToPrefix("/bar/foo/")
'bar_foo_'
```

It also seems that I can replicate all the examples with a much simpler piece of code:
```python
def repoNameToPrefix(repo):
    return repo.strip(os.sep).replace(os.sep, "_")
```
Is there a subtlety I'm missing?